// @4c/layout/lib/PropTypes.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.defaultProps = exports.propTypes = void 0;

var _propTypes = _interopRequireDefault(require("prop-types"));

var _elementType = _interopRequireDefault(require("prop-types-extra/lib/elementType"));

const align = _propTypes.default.oneOf(['start', 'end', 'flex-start', 'flex-end', 'center', 'stretch', 'baseline', 'first-baseline', 'last-baseline']);

const content = _propTypes.default.oneOf(['left', 'right', 'flex-start', 'flex-end', 'center', 'baseline', 'first-baseline', 'last-baseline', 'space-between', 'space-around', 'space-evenly']);

const propTypes = {
  direction: _propTypes.default.oneOf(['row', 'column']).isRequired,
  as: _elementType.default.isRequired,
  pad: _propTypes.default.oneOfType([_propTypes.default.bool, _propTypes.default.number]),
  wrap: _propTypes.default.bool,
  grow: _propTypes.default.bool,
  inline: _propTypes.default.bool,
  flex: _propTypes.default.any,
  align: align.isRequired,
  alignSelf: align,
  alignContent: content,
  justify: content.isRequired
};
exports.propTypes = propTypes;
const defaultProps = {
  as: 'div',
  direction: 'row',
  align: 'stretch',
  justify: 'flex-start'
};
exports.defaultProps = defaultProps;
